examples/html-ak-basic-sign-client/src/main.js

```diff
@@ -111,19 +200,101 @@ function setupEventListeners() {
     updateDom()
   })
 
-  document.getElementById('sign-message')?.addEventListener('click', async () => {
-    await signClient.request({
-      topic: session.topic,
-      chainId: 'eip155:1',
-      request: {
-        method: 'personal_sign',
-        params: [
-          '0x7468697320697320612074657374206d65737361676520746f206265207369676e6564',
-          account
-        ]
+  document.getElementById('sign-message')?.addEventListener('click', signMessage)
+}
+
+async function getPayload() {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const map = {
+    solana: {
+      method: 'solana_signMessage',
+      params: {
+        message: base58.encode(new TextEncoder().encode('Hello Appkit!')),
+        pubkey: account
+      }
+    },
+    eip155: {
+      method: 'personal_sign',
+      params: [account, 'Hello AppKit!']
+    },
+    bip122: {
+      method: 'signMessage',
+      params: {
+        message: 'Hello AppKit!',
+        account: account
       }
+    },
+    polkadot: {
+      method: 'polkadot_signMessage',
+      params: {
+        transactionPayload: {
+          specVersion: '0x00002468',
+          transactionVersion: '0x0000000e',
+          address: `${account}`,
+          blockHash: '0x554d682a74099d05e8b7852d19c93b527b5fae1e9e1969f6e1b82a2f09a14cc9',
+          blockNumber: '0x00cb539c',
+          era: '0xc501',
+          genesisHash: '0xe143f23803ac50e8f6f8e62695d1ce9e4e1d68aa36c1cd2cfd15340213f3423e',
+          method: '0x0001784920616d207369676e696e672074686973207472616e73616374696f6e21',
+          nonce: '0x00000000',
+          signedExtensions: [
+            'CheckNonZeroSender',
+            'CheckSpecVersion',
+            'CheckTxVersion',
+            'CheckGenesis',
+            'CheckMortality',
+            'CheckNonce',
+            'CheckWeight',
+            'ChargeTransactionPayment'
+          ],
+          tip: '0x00000000000000000000000000000000',
+          version: 4
+        },
+        address: account
+      }
+    }
+  }
+
+  const [namespace] = network.split(':')
+  const payload = map[namespace || '']
+
+  return payload
+}
+
+async function signMessage() {
+  try {
+    if (!signClient || !account) {
+      throw Error('User is disconnected')
+    }
+
+    const payload = await getPayload()
+
+    if (!payload) {
+      throw Error('Chain not supported by laboratory')
+    }
+
+    const sig = await signClient.request({
+      topic: session.topic,
+      chainId: network,
+      request: payload
     })
-  })
+
+    console.log({
```

Comment:
Log

Reply:
This is on puirpose to log the signature